### PR TITLE
Feat: Buttons - Adjust notification, message content, and options menu for composite …

### PR DESCRIPTION
**Pull request by gizemb**
…message

Jira tickets:
https://wearezeta.atlassian.net/browse/AN-6734
https://wearezeta.atlassian.net/browse/AN-6729

## What's new in this PR?

This PR adds support for
- Displaying proper notification text
- Message summary text in conversation list
- Searching contents of the message
- Displaying message options menu
- Handling click actions
- Read receipts

for Composite messages

### Testing
Manually tested since we don't have a testing infrastructure on the UI side.

### CODE REVIEW

File: app/src/main/res/layout/layout_composite_message.xml [hunk 1]
```diff
@@ -6,17 +6,15 @@
...
     tools:parentTag="android.widget.LinearLayout">
 
-    <LinearLayout
-        android:id="@+id/composite_message_text_container"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_marginBottom="@dimen/composite_message_text_layout_margin_bottom"
-        android:orientation="vertical" />
+    <include
+        android:id="@+id/composite_message_textpartview"
```
Comment (BradleyWire) [inline app/src/main/res/layout/layout_composite_message.xml]: This should be `composite_message_text_partview`

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/MessageData.scala [hunk 2]
```diff
@@ -70,14 +68,10 @@ case class MessageData(override val id:   MessageId              = MessageId(),
...
                       ) extends Identifiable[MessageId] with DerivedLogTag {
-  def getContent(index: Int) = {
-    if (index == 0) content.headOption.getOrElse(MessageContent.Empty)
-    else content.drop(index).headOption.getOrElse(MessageContent.Empty)
-  }
-
   lazy val contentString = protos.lastOption match {
     case Some(TextMessage(ct, _, _, _, _)) => ct
     case _ if msgType == api.Message.Type.RICH_MEDIA => content.map(_.content).mkString(" ")
+    case _ if msgType == api.Message.Type.COMPOSITE => content.map(_.content).mkString("\n")
```
Comment (BradleyWire) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/MessageData.scala]: Formatting here with the `=>`

File: app/src/main/res/layout/layout_composite_message.xml (hunk 1 above)
Comment (gizemb) [reply app/src/main/res/layout/layout_composite_message.xml]: the widget's name is `TextPartView` so I kept it like this.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/MessageData.scala (hunk 2 above)
Comment (gizemb) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/MessageData.scala]: I'll fix it if I push another patch set 👍

**Pull request merged**